### ISSUE TITLE
Possible rounding error when rendering SVG to PNG

### ISSUE THREAD

**Issue by m1h4** [lovell/sharp#1947]
When trying to render the following svg to png:

```
<?xml version="1.0"?>
<svg xmlns="http://www.w3.org/2000/svg" style="vertical-align: -0.05ex" width="3.491ex" height="2.397ex" role="img" focusable="false" viewBox="0 -1037.5 1543 1059.5">
    <g stroke="currentColor" fill="currentColor" stroke-width="0" transform="matrix(1 0 0 -1 0 0)">
        <g data-mml-node="math">
            <g data-mml-node="mover">
                <g data-mml-node="mrow">
                    <g data-mml-node="mi">
                        <path data-c="43" d="M50 252Q50 367 117 473T286 641T490 704Q580 704 633 653Q642 643 648 636T656 626L657 623Q660 623 684 649Q691 655 699 663T715 679T725 690L740 705H746Q760 705 760 698Q760 694 728 561Q692 422 692 421Q690 416 687 415T669 413H653Q647 419 647 422Q647 423 648 429T650 449T651 481Q651 552 619 605T510 659Q484 659 454 652T382 628T299 572T226 479Q194 422 175 346T156 222Q156 108 232 58Q280 24 350 24Q441 24 512 92T606 240Q610 253 612 255T628 257Q648 257 648 248Q648 243 647 239Q618 132 523 55T319 -22Q206 -22 128 53T50 252Z"/>
                    </g>
                    <g data-mml-node="mi" transform="translate(715, 0)">
                        <path data-c="44" d="M287 628Q287 635 230 637Q207 637 200 638T193 647Q193 655 197 667T204 682Q206 683 403 683Q570 682 590 682T630 676Q702 659 752 597T803 431Q803 275 696 151T444 3L430 1L236 0H125H72Q48 0 41 2T33 11Q33 13 36 25Q40 41 44 43T67 46Q94 46 127 49Q141 52 146 61Q149 65 218 339T287 628ZM703 469Q703 507 692 537T666 584T629 613T590 629T555 636Q553 636 541 636T512 636T479 637H436Q392 637 386 627Q384 623 313 339T242 52Q242 48 253 48T330 47Q335 47 349 47T373 46Q499 46 581 128Q617 164 640 212T683 339T703 469Z"/>
                    </g>
                </g>
                <g data-mml-node="mo" transform="translate(0, 520.3) scale(0.707)">
                    <svg width="2182.1" height="246" x="0" y="444" viewBox="545.5 444 2182.1 246">
                        <path data-c="AF" d="M69 544V590H430V544H69Z" transform="scale(6.546, 1)"/>
                    </svg>
                </g>
            </g>
        </g>
    </g>
</svg>
```

The image produced is cut at the bottom:

![test](https://user-images.githubusercontent.com/1467554/67960532-b8109100-fbfa-11e9-9f9b-31d103c42c1e.png)

In contrast, the following file:

```
<?xml version="1.0"?>
<svg xmlns="http://www.w3.org/2000/svg" style="vertical-align: 0" width="3.59ex" height="2.298ex" role="img" focusable="false" viewBox="0 -1015.5 1587 1015.5">
    <g stroke="currentColor" fill="currentColor" stroke-width="0" transform="matrix(1 0 0 -1 0 0)">
        <g data-mml-node="math">
            <g data-mml-node="mover">
                <g data-mml-node="mrow">
                    <g data-mml-node="mi">
                        <path data-c="42" d="M231 637Q204 637 199 638T194 649Q194 676 205 682Q206 683 335 683Q594 683 608 681Q671 671 713 636T756 544Q756 480 698 429T565 360L555 357Q619 348 660 311T702 219Q702 146 630 78T453 1Q446 0 242 0Q42 0 39 2Q35 5 35 10Q35 17 37 24Q42 43 47 45Q51 46 62 46H68Q95 46 128 49Q142 52 147 61Q150 65 219 339T288 628Q288 635 231 637ZM649 544Q649 574 634 600T585 634Q578 636 493 637Q473 637 451 637T416 636H403Q388 635 384 626Q382 622 352 506Q352 503 351 500L320 374H401Q482 374 494 376Q554 386 601 434T649 544ZM595 229Q595 273 572 302T512 336Q506 337 429 337Q311 337 310 336Q310 334 293 263T258 122L240 52Q240 48 252 48T333 46Q422 46 429 47Q491 54 543 105T595 229Z"/>
                    </g>
                    <g data-mml-node="mi" transform="translate(759, 0)">
                        <path data-c="44" d="M287 628Q287 635 230 637Q207 637 200 638T193 647Q193 655 197 667T204 682Q206 683 403 683Q570 682 590 682T630 676Q702 659 752 597T803 431Q803 275 696 151T444 3L430 1L236 0H125H72Q48 0 41 2T33 11Q33 13 36 25Q40 41 44 43T67 46Q94 46 127 49Q141 52 146 61Q149 65 218 339T287 628ZM703 469Q703 507 692 537T666 584T629 613T590 629T555 636Q553 636 541 636T512 636T479 637H436Q392 637 386 627Q384 623 313 339T242 52Q242 48 253 48T330 47Q335 47 349 47T373 46Q499 46 581 128Q617 164 640 212T683 339T703 469Z"/>
                    </g>
                </g>
                <g data-mml-node="mo" transform="translate(0, 498.3) scale(0.707)">
                    <svg width="2244.4" height="246" x="0" y="444" viewBox="561.1 444 2244.4 246">
                        <path data-c="AF" d="M69 544V590H430V544H69Z" transform="scale(6.733, 1)"/>
                    </svg>
                </g>
            </g>
        </g>
    </g>
</svg>
```

is not:

![test](https://user-images.githubusercontent.com/1467554/67960427-87c8f280-fbfa-11e9-9ea3-8199dc3bfcb5.png)

Seems to possibly be a rounding issue when determining canvas size/height.

The above images are generated with `sharp('file.svg', {density: 220}).toFile('file.png')`.

**Comment by lovell** [comment]
Thanks for reporting, this looks like a bug in libvips. I've created https://github.com/libvips/libvips/pull/1459 to fix it.

**Comment by m1h4** [comment]
Thank you for a very fast response!

Sorry it ended up being a duplicate in upstream…

**Comment by lovell** [comment]
v0.24.0 is now available with a prebuilt libvips v8.9.0.
